### PR TITLE
[s] Stops infinite food generation by frying the same item repeatedly

### LINKED ISSUE #38025: Fried monkey hides spontaneously become un-fried when eaten
[Round ID]: # 88400

[Testmerges]: # #37867: 'Adds an admin punishment: Spewapalooza' by Armhulen at commit b13873b73e 

[Reproduction]: # Obtain a monkey hide, fry the monkey hide (the ones I ate were lightly-fried but it probably works with any level of fried), eat the monkey hide. The monkey hide will drop at your feet, un-fried, after the first bite. You can fry this hide and repeat this infinitely.

### PR DESCRIPTION
**Pull request by ShizCalev**
Closes #38025
Closes #38054

🆑 ShizCalev
fix: You can now only deepfry a single item once.
/🆑

### CODE REVIEW

**Comment by Iamgoofball** [comment]
This shouldn't need to happen, the doublefry blocker should stop it

**Comment by ShizCalev** [comment]
@Iamgoofball When you fry a non-food item and eat it, it drops the original item. You can then fry it again afterward, allowing you to boost your mood infinitely.

**Comment by Iamgoofball** [comment]
just make it delete the item on frying, keeping it is stupid and was only added to appease immersion nerds

**Comment by XDTM** [comment]
Sugar comes from chem dispensers, gives nutrition and is near infinite (no satiety limit either). This is less problematic in comparison.

**Comment by Fox-McCloud** [comment]
>just make it delete the item on frying, keeping it is stupid and was only added to appease immersion nerds

Yeah, I dunno why the original item isn't just deleted; this would completely solve the problem.

**Comment by SgtxRootbeer** [comment]
Traitor items/nuke disks, etc.

**Comment by PKPenguin321** [comment]
This is intended and this fix is stupid. You can fry literally anything and eat it, including your clothes. Preventing somebody from frying the same pair of shoes a second time won't stop them from just frying and eating some random shit they found on the floor instead.

**Comment by Therapist1** [comment]
i don’t see the point. do this and they’ll just walk over with a paper bin or card deck, take one out, fry it, discard it, and fry another

**Comment by SgtxRootbeer** [comment]
tbh, as many have pointed out, this doesnt make sense and isnt an issue. why cant we stick something that we ate the breading off of back into the fryer? by then its surely not double-frying. 
that and if people want to eat, they're going to eat.

**Comment by davethwave** [comment]
What makes frying a item thats been fried before after eating it horrible enough to require a fix PR? Is it even a problem in the first place? Your logic for this seems to be that it closes one of the issue's however it seems to me that this wouldn't even solve the problem, making the reasoning for this faulty.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -106,6 +106,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	//Grinder vars
 	var/list/grind_results //A reagent list containing the reagents this item produces when ground up in a grinder - this can be an empty list to allow for reagent transferring only
 	var/list/juice_results //A reagent list containing blah blah... but when JUICED in a grinder!
+	var/was_previously_fried //Prevents the same item from being refried multiple times
```
Comment (vuonojenmustaturska) [inline code/game/objects/items.dm]: booleans like these belong in item_flags imo

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

**Pull request closed without merging**